src/dict/dictserver.cc

```diff
@@ -538,14 +538,22 @@ void DictServerWordSearchRequest::readMatchData( QByteArray & reply )
       if ( word.endsWith( '\"' ) ) {
         word.chop( 1 );
       }
-      if ( word[ 0 ] == '\"' ) {
+      if ( word.startsWith( '\"' ) ) {
         word = word.mid( 1 );
       }
 
-      this->addMatchedWord( word );
+      if ( !word.isEmpty() ) {
+        this->addMatchedWord( word );
+      }
     }
 
-    reply = this->dictImpl->socket.readLine();
+    constexpr int halfSecond = 500;
+    if ( this->dictImpl->socket.waitForReadyRead( halfSecond ) ) {
+      reply = this->dictImpl->socket.readLine();
+    }
```

Comment:
IMO, waitForReadyRead will block the socket and make the GUI wait.

Reply:
It right now straight to crash 😅.

Reply:
I will just fix the string part first.